src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java

```diff
@@ -0,0 +1,139 @@
+                                        @Value("${config.taskMandatoryFields}") List<String> taskMandatoryFields) {
+        this.launchDarklyFeatureFlagProvider = launchDarklyFeatureFlagProvider;
+        this.isMandatoryFieldCheckEnabled = taskMandatoryFieldCheckEnabled;
+        this.taskMandatoryFields = taskMandatoryFields;
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     */
+    public void validate(TaskResource task) {
+        log.info("Validating mandatory fields for task {}", task.getTaskId());
+
+        if (isMandatoryFieldCheckEnabled) {
+            LDValue mandatoryFieldCheckEnabledServices = launchDarklyFeatureFlagProvider.getJsonValue(
+                FeatureFlag.WA_MANDATORY_FIELD_CHECK,
+                "ccd-case-disposer",
+                "ccd-case-disposer@hmcts.net",
+                LDValue.of("{\"jurisdictions\": []}")
+            );
+
+            if (mandatoryFieldCheckEnabledServices == null) {
+                log.warn("Mandatory field check enabled services is null, skipping validation.");
+                return;
+            }
+
+            JsonNode excludedJurisdictionsArray = parseJson(mandatoryFieldCheckEnabledServices.toJsonString());
+            if (excludedJurisdictionsArray == null) {
+                log.warn("Excluded jurisdictions array is null, skipping jurisdiction check.");
+                validateTaskMandatoryFields(task);
+                return;
+            }
+
+            if (!isJurisdictionExcluded(task, excludedJurisdictionsArray)) {
+                validateTaskMandatoryFields(task);
+            }
+        }
+    }
+
+    /**
+     * Checks if the task's jurisdiction is excluded.
+     *
+     * @param task the task to be checked
+     * @param excludedJurisdictionsArray the array of excluded jurisdictions
+     * @return true if the jurisdiction is excluded, false otherwise
+     */
+    private boolean isJurisdictionExcluded(TaskResource task, JsonNode excludedJurisdictionsArray) {
+        for (JsonNode jurisdiction : excludedJurisdictionsArray) {
+            if (jurisdiction.asText().equals(task.getJurisdiction())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Parses a JSON string and returns the JsonNode for the mandatory field check flag variant.
+     *
+     * @param jsonString the JSON string to be parsed
+     * @return the JsonNode for the mandatory field check flag variant
+     * @throws IllegalArgumentException if the JSON string cannot be parsed
+     */
+    public JsonNode parseJson(String jsonString) {
+        try {
+            return new ObjectMapper().readTree(jsonString).get(MANDATORY_FIELD_CHECK_FLAG_VARIANT.stringValue());
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("Mandatory flag jurisdictions mapping issue.", e);
+        }
+    }
```

Comment:
Maybe this could be in another class (JsonParser if not one already?) and made more generic  e.g. public JsonNode getJsonNode(String jsonString, String nodeName) ?  

Also a minor consideration should we be using the ObjectMapper from the SpringContext or is there a problem with that?